src/core/src/sketch/minhash.rs

```diff
@@ -435,19 +432,8 @@ impl KmerMinHash {
             let mut self_iter = self.mins.iter();
             let mut other_iter = other.mins.iter();
 
-            let mut self_abunds_iter: Option<std::slice::Iter<'_, u64>>;
-            if let Some(ref mut abunds) = self.abunds {
-                self_abunds_iter = Some(abunds.iter());
-            } else {
-                self_abunds_iter = None;
-            }
-
-            let mut other_abunds_iter: Option<std::slice::Iter<'_, u64>>;
-            if let Some(ref abunds) = other.abunds {
-                other_abunds_iter = Some(abunds.iter());
-            } else {
-                other_abunds_iter = None;
-            }
+            let mut self_abunds_iter = self.abunds.as_mut().map(|a| a.iter());
```

Comment:
Hi Luiz,
This is just a question.
So, `self_abunds_iter` type was `Option` that can contain either of two values. Will the new iterator hold one of the two types `iter` or `None`?

Reply:
Yup! `self.abunds` is an `Option<Vec<u64>>`, so [.as_mut()](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_mut) requests an `Option<&mut Vec<u64>` and finally [`.map`](https://doc.rust-lang.org/std/option/enum.Option.html#method.map) allows us to call a method (`.iter()`) on `&mut Vec<u64>`, which gives our final type: `Option<std::slice::Iter<'_, u64>>` (the same as before).

This was two different clippy suggestion combined, so it might look a bit mystical at first...

Reply:
I liked it! Thanks, Luiz!